x-pack/test/functional/apps/machine_learning/data_visualizer/index_data_visualizer.ts

```diff
@@ -0,0 +1,237 @@
+            type: 'keyword',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+        ],
+        nonMetricFieldsTypeFilterCardCount: 3,
+        metricFieldsFilterCardCount: 1,
+        nonMetricFieldsFilterCardCount: 1,
+      },
+    },
+    {
+      suiteTitle: 'with lucene query on farequote index',
+      sourceIndexOrSavedSearch: 'farequote_lucene',
+      metricFieldsFilter: 'responsetime',
+      nonMetricFieldsFilter: 'version',
+      nonMetricFieldsTypeFilter: 'keyword',
+      expected: {
+        totalDocCount: 34416,
+        fieldsPanelCount: 2, // Metrics panel and Fields panel
+        metricCards: [
+          {
+            type: 'number', // document count card
+          },
+          {
+            fieldName: 'responsetime',
+            type: 'number',
+          },
+        ],
+        nonMetricCards: [
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+          {
+            fieldName: '@version',
+            type: 'text',
+          },
+          {
+            fieldName: '@version.keyword',
+            type: 'keyword',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+          {
+            fieldName: '@timestamp',
+            type: 'date',
+          },
+        ],
+        nonMetricFieldsTypeFilterCardCount: 3,
+        metricFieldsFilterCardCount: 2,
+        nonMetricFieldsFilterCardCount: 1,
+      },
+    },
+  ];
+
+  describe('index based', function() {
+    this.tags(['smoke', 'mlqa']);
+    before(async () => {
+      await esArchiver.load('ml/farequote');
+    });
+
+    after(async () => {
+      await esArchiver.unload('ml/farequote');
+    });
+
+    // TODO - add tests for
+    //  - validating metrics displayed inside the cards
+    //  - selecting a document sample size
+    //  - clicking on the link to the Advanced job wizard
+    for (const testData of testDataList) {
+      describe(`${testData.suiteTitle}`, function() {
+        it('loads the data visualizer selector page', async () => {
+          await ml.navigation.navigateToMl();
+          await ml.navigation.navigateToDataVisualizer();
+        });
+
+        it('loads the saved search selection page', async () => {
+          await ml.dataVisualizer.navigateToIndexPatternSelection();
+        });
+
+        it('loads the index data visualizer page', async () => {
+          await ml.jobSourceSelection.selectSourceForIndexBasedDataVisualizer(
+            testData.sourceIndexOrSavedSearch
+          );
+        });
+
+        it('index data visualizer displays the time range step', async () => {
```

Comment:
The `index data visualizer` part of this name is redundant. The whole test name is concatenated from all surrounding suites, so for this one it would be `machine learning data visualizer index based with lucene query on farequote index index data visualizer displays the time range step`. I would suggest to remove `index data visualizer` from this and the following test names.

Reply:
Done in https://github.com/elastic/kibana/pull/51832/commits/dfde35a493e596fe1db4d22b8d6dc00e0d14de52